Code/GraphMol/ChemTransforms/MolZip.cpp

```diff
@@ -334,6 +335,73 @@ struct ZipBond {
     }
   }
 };
+
+void rotateFragmentToBondVector(
+    ROMol &mol, const Atom &a, const Atom &b, const Atom &a_dummy,
+    const Atom &b_dummy, const std::vector<int> &fragmentForAtom,
+    const std::map<int, std::vector<int>> &atomsInFragment, int confId = -1) {
+  if (!mol.getNumConformers()) {
+    return;
+  }
+  auto &conf = mol.getConformer(confId);
+
+  //-----------------------------------------
+  //  Notation:
+  //    Pmc: molecule connection point (the atom that will be
+  //     removed from the molecule).
+  //    Pma: molecule attachment point (the atom to which we'll form
+  //     the bond).
+  //    Psc: sidechain connection point
+  //    Psa: sidechain attachment point
+  //    Vm: Pmc-Pma (molecular attachment vector)
+  //    Vs: Psc-Psa (sidechain attachment vector)
+  //
+  //-----------------------------------------
+  const auto Pma = conf.getAtomPos(a.getIdx());
+  const auto Psa = conf.getAtomPos(b.getIdx());
+  const auto Pmc = conf.getAtomPos(a_dummy.getIdx());
+  const auto Psc = conf.getAtomPos(b_dummy.getIdx());
+
+  auto Um = Pma.directionVector(Pmc);
+  // note the opposite direction here:
+  auto Us = Psc.directionVector(Psa);
+
+  RDGeom::Transform3D templateTform;
+  templateTform.SetTranslation(Pma);
+
+  double sinT, cosT;
+  cosT = Us.dotProduct(Um);
+  if (cosT > 1.0) cosT = 1.0;
+  if (fabs(cosT) < 1.0) {
+    sinT = sqrt(1.0 - cosT * cosT);
+    RDGeom::Point3D rotnAxis = Us.crossProduct(Um);
+    rotnAxis.normalize();
+    templateTform.SetRotation(cosT, sinT, rotnAxis);
+  } else if (cosT == 1.0) {
+    RDGeom::Point3D normal(1, 0, 0);
+    if (fabs(Us.dotProduct(normal)) == 1.0) {
+      normal = RDGeom::Point3D(0, 1, 0);
+    }
+    RDGeom::Point3D rotnAxis = Us.crossProduct(normal);
+    templateTform.SetRotation(-1, 0, rotnAxis);
+  }
+
```

Comment:
I don't think the geometry is correct here.  For example,
```
  SECTION("on top") {
    auto m3 = "[1*]c1ccccc1.[1*]c1ncccc1 |(2.57143,0,;1.07143,0,;0.321429,-1.29904,;-1.17857,-1.29904,;-1.92857,0,;-1.17857,1.29904,;0.321429,1.29904,;12.5714,0,;11.0714,0,;10.3214,1.29904,;8.82143,1.29904,;8.07143,0,;8.82143,-1.29904,;10.3214,-1.29904,)|"_smiles;
    MolzipParams params;
    params.alignCoordinates = true;
    params.label = MolzipLabel::Isotope;
    auto res = molzip(*m3, params);
    std::cout << MolToCXSmiles(*res) << std::endl;
  }
```
gives

<img width="600" height="200" alt="Image" src="https://github.com/user-attachments/assets/1e71dbf7-5df5-4870-8fc9-2e73c4012170" />

Reply:
I have several decades of experience in getting this sort of thing wrong, so I know what to look out for :-).